fc/protocol.py

```diff
@@ -445,53 +439,30 @@ def outputs_and_plots(self, errors, verbose=True, write_out=True):
         # Plots
         if write_out:  # suppress plotting when performing fitting
             start = time.time()
-            for plot in self.plots:
-                with errors:
+            with errors:
+                for plot in self.plots:
+                    path = os.path.join(self.output_folder.path, self.sanitise_file_name(plot['title']) + '.png')
+                    x_label = plot_descriptions[plot['x']]
+                    y_label = plot_descriptions[plot['y']]
+                    x_data = [self.output_env.look_up(plot['x']).array]
+                    y_data = [self.output_env.look_up(plot['y']).array]
+
                     if verbose:
                         self.log_progress(
                             'plotting', plot['title'],
                             'curve:', plot_descriptions[plot['y']],
                             'against', plot_descriptions[plot['x']]
                         )
-                    x_data = []
-                    y_data = []
-                    x_data.append(self.output_env.look_up(plot['x']).array)
-                    y_data.append(self.output_env.look_up(plot['y']).array)
+
+                    # TODO Looks like this isn't used?
```

Comment:
Indeed, it doesn't go in the .png, but having it here means we check for the error case.